src/state/session/index.tsx

```diff
@@ -46,29 +46,67 @@ const ApiContext = React.createContext<SessionApiContext>({
 })
 ApiContext.displayName = 'SessionApiContext'
 
-export function Provider({children}: React.PropsWithChildren<{}>) {
-  const cancelPendingTask = useOneTaskAtATime()
-  const [state, dispatch] = React.useReducer(reducer, null, () => {
+class SessionStore {
+  private state: State
+  private listeners = new Set<() => void>()
+
+  constructor() {
+    // Careful: By the time this runs, `persisted` needs to already be filled.
     const initialState = getInitialState(persisted.get('session').accounts)
```

Comment:
What happens in the case where there is no persisted session or no accounts? Could this result in an error?

Reply:
The `persisted` abstraction fills that in with default values (i.e. we get `[]` here). This part is similar to how it worked before, I just moved it to a class.

Reply:
The timing caveat is just that the initial read is async. We delay the component tree from mounting until then but reading in module scope would've read too early.